samples/snippets/connection_create.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# [START bigqueryconnection_connection_create]
+from google.cloud import bigquery_connection_v1 as bq_connection
+from google.cloud.bigquery_connection_v1 import types as connection_types
+
+"""This sample shows how to create a BigQuery connection with a Cloud MySql database"""
+
+
+def main(
+    project_id: str,
+    location: str,
+    connection_id: str,
+    database: str,
+    instance: str,
+    instance_location: str,
+    username: str,
+    password: str,
+) -> connection_types.Connection:
+    # [START_EXCLUDE]
+    original_project_id = project_id
+    original_location = location
+    original_connection_id = connection_id
+    original_database = database
+    original_instance = instance
+    original_instance_location = instance_location
+    original_username = username
+    original_password = password
+    # [END_EXCLUDE]
```

Comment:
What's up with these excludes and reassignments? Can we just leave this as is so the suer can edit it, and test the behavior of the function below?

Reply:
This is a pattern we've been following recently - I believe the purpose is to allow the function to be "live" and testable while still giving a clear view to the developer on how to fill in their own values. To me I think it makes more sense to have developer values set as statics above the function itself, but I'm not sure if that's consistent with what we've been doing. @tswast may have an opinion!

Reply:
The `EXCLUDE` pattern is probably not necessary if the snippet includes the whole function definition, as you've done here. Using default values there would serve the same purpose.

Reply:
Also, I tend to avoid `EXCLUDE` even with this pattern, as it shows "..." when rendered in the docs.

Reply:
https://github.com/googleapis/python-datacatalog/blob/main/samples/snippets/data_catalog_ptm_create_taxonomy.py is a more typical example.

In core BigQuery we have a lot of samples that have the tags within the function, so we've kept with that style.

Reply:
Slight update -- I've started to use the more standard style in some BigQuery samples: https://github.com/googleapis/python-bigquery/pull/1137

Reply:
Attempted in recent commit! Would love feedback!